applications/serial_lte_modem/doc/PPP_AT_commands.rst

```diff
@@ -0,0 +1,95 @@
+.. _SLM_AT_PPP:
+
+PPP AT commands
+****************
+
+.. contents::
+   :local:
+   :depth: 2
+
+This page describes PPP-related AT commands.
+
+Control PPP #XPPP
+=================
+
+Set command
+-----------
+
+The set command allows you to start and stop PPP.
+
+Syntax
+~~~~~~
+
+::
+
+   #XPPP=<op>
+
+* The ``<op>`` parameter can be the following:
+
+  * ``0`` - Stop PPP.
+  * ``1`` - Start PPP.
+    If PPP is already started, it is first stopped before being started again.
+
+.. note::
+
+   PPP is automatically started and stopped when the default PDN connection is established and lost, respectively.
+   This happens even if PPP has previously been stopped or started with this command.
+
+Read command
+------------
+
+The read command allows you to get the status of PPP.
+
+Syntax
+~~~~~~
+
+::
+
+   #XPPP?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XPPP: <running>,<peer_connected>
+
+* The ``<running>`` parameter is an integer that indicates whether PPP is running.
+  It is ``1`` for running or ``0`` for stopped.
+
+* The ``<peer_connected>`` parameter is an integer that indicates whether a peer is connected to PPP.
+  It is ``1`` for connected or ``0`` for not connected.
+
+Test command
+------------
+
+The test command is not supported.
+
+Example
+-------
+
+::
+
+  AT+CFUN=1
+
+  OK
+  // Wait for network registration. PPP is automatically started.
+  AT#XPPP?
+
+  #XPPP: 1,0
+
+  OK
+  // Connect to PPP. The read command shows that a peer is connected.
```

Comment:
From SLM's perspective yes, but from the user doing this not necessarily, e.g. here you would start `pppd`.

Reply:
Should this example then include the AT#PPP=1?

Reply:
No. The start happens indeed automatically on the SLM side with the changes from this PR. PPP will get auto-started when the network registration happens, which in this example is between `AT+CFUN=1` and the first `AT#XPPP?`.

Reply:
OK, so it is as I first thought. "Connect to PPP." to me reads as an action that you would do. So that is why I thought that this was missing a command, with that phrasing.

Reply:
I mean yes, that is what I was subtly trying to communicate. At that point you may want to start `pppd`, for example. But that may need to be made more explicit. How about something like below?
```suggestion
  // Have the peer connect to SLM's PPP. The read command shows that a peer is connected.
```

Reply:
Ah yes, forgot the starting of the pppd at this point and thus had problems in understanding this. This is definitively better.

Not on the scope for this, but should we document the pppd command as an example? It would aid in trying this out with Linux, but then there is the question of Windows usage...

Reply:
Regarding additional documentation, yes definitely. This is planned.